### ISSUE TITLE
Slice gives incorrect type for template literal types like `'foo${string}'`

### ISSUE THREAD

**Issue by dd-jonas** [gustavoguichard/string-ts#111]
Given a variable of type `'foo${string}'`, `slice(s, 0, 1)`, `slice(s, 0, 2)` and `slice(s, 0, 3)` correctly infer the types `'f'`, `'fo'`and `'foo'` respectively. However, when using an end value higher than the length of the literal part of the string, the inferred type remains `'foo'` (e.g. for `slice(s, 0, 6)`). Same goes for e.g. `slice(s, -4)` if the literal part is at the end of the string. It gets even weirder with e.g. `slice(s, 0, 7)` in case `s` is of type `'foo${string}${bar}'`

Not sure if this is known or fixable behaviour.

**Comment by gustavoguichard** [comment]
Hey @dd-jonas , it is a known issue and it was fixed in #105 
Not released yet, but I'm going to.

What we are doing in this case is to return `string` instead of a literal. It is too complex to do otherwise so we'll try to keep the scope to only literal and unions of literals, that means:
```ts
type A = Slice<'foo', 0, 2> // 'fo'
type B = Slice<'foo' | 'bar', 0, 2> // 'fo' | 'ba'
type C = Slice<`foo${string}`, 0, 2> // string
type D = Slice<'foo', 0, number> // string
```

So you wouldn't be able to work with that interpolation but at least you won't have weird behavior.

Keep an eye out for the next release.

**Comment by gustavoguichard** [comment]
@dd-jonas this issue should now be solved in v1.3.3 .
